src/model.cpp

```diff
@@ -219,4 +233,129 @@ size_t Model::unitsCount() const
     return mPimpl->mUnits.size();
 }
 
+typedef std::shared_ptr<ImportedEntity> ImportedEntityPtr;
+
+/**
+ * @brief Resolve the path of the given filename using the given base.
+ *
+ * Resolves the full path to the given @p filename using the @p base.
+ *
+ * This function is only intended to work with local files.  It may not
+ * work with bases that use the 'file://' prefix.
+ *
+ * @param filename The @c std::string relative path from the base path.
+ * @param base The @c std::string location on local disk for determining the full path from.
+ * @return The full path from the @p base location to the @p filename
+ */
+std::string resolvePath(const std::string &filename, const std::string &base)
+{
+    // We can be naive here as we know what we are dealing with
+    std::string path = base.substr(0, base.find_last_of('/')+1) + filename;
+    return path;
+}
+
+void resolveImport(ImportedEntityPtr importedEntity,
+                   const std::string &baseFile)
+{
+    if (importedEntity->isImport()) {
+        libcellml::ImportSourcePtr importSource = importedEntity->getImportSource();
+        if (!importSource->hasModel()) {
+            std::string url = resolvePath(importSource->getUrl(), baseFile);
+            std::ifstream file(url);
+            if (file.good()) {
+                std::stringstream buffer;
+                buffer << file.rdbuf();
+                libcellml::Parser parser;
+                libcellml::ModelPtr model = parser.parseModel(buffer.str());
+                if (model) {
+                    importSource->setModel(model);
+                    model->resolveImports(url);
```

Comment:
Won't this cause all imports in the imported model to be resolved, regardless of whether they are needed?

Reply:
OK, so we always try to resolve all imports in all imported models, but the `hasUnresolvedImports` method only checks the required imports are resolved. I guess this is more an efficiency thing that an actual problem with the code.

Reply:
My point was that, from the code in `Parser::parseModel()`, `model` will never be equal to `nullptr`, so no point for the test.